proparse/src/main/java/org/prorefactor/treeparser/ParseUnit.java

```diff
@@ -126,29 +116,45 @@ public File getFile() {
 
   public TokenStream lex() throws RefactorException {
     LOGGER.trace("Entering ParseUnit#lex()");
-    DoParse doParse = new DoParse(session, file.getPath());
     try {
-      doParse.doParse(true);
-    } catch (ANTLRException | IOException caught) {
+      ProgressLexer lexer = new ProgressLexer(session, file.getPath());
+      TokenStream stream = lexer.getANTLR2TokenStream(false);
+      try {
+        Token tok = stream.nextToken();
+        while (tok.getType() != Token.EOF_TYPE) {
+          tok = stream.nextToken();
+        }
+      } catch (TokenStreamException uncaught) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS00108)